### ISSUE TITLE
Question: Is there a dependency on Python 2?

### ISSUE THREAD

**Issue by Croydon** [asbru-cm/asbru-cm#354]
There seems to be a direct or indirect dependency on Python2

On my Fedora 31 installation where I have installed Asbru via the packagecloud repository for Fedora 31 I can't get rid of the `python2` package without uninstalling asbru-cm

Can you confirm that?

**Comment by gfrenoy** [comment]
You can check actual dependencies here:

https://packagecloud.io/asbru-cm/asbru-cm/packages/fedora/31/asbru-cm-6.0.2-1.fc31.noarch.rpm

(also given by ```rpm -qpR asbru-cm-6.0.2-1.fc31.noarch.rpm```).

There is for sure no direct dependency.

If you want to check on your environment, try this (when Ásbrú is installed):

```
rpm -qR asbru-cm
```

**Comment by gfrenoy** [comment]
I would think this is the ```packagecloud.io``` [install script](https://packagecloud.io/asbru-cm/asbru-cm/install#bash-rpm) that is requiring python2 ; in particular, they are installing ```pygpgme```.

If you really don't want to have python2, I believe you can remove ```pygpgme``` and then set ```repo_gpgcheck=0``` in ```/etc/yum.repos.d/asbru-cm_asbru-cm.repo``` ; of course no GPG verification will be done.

**Comment by Croydon** [comment]
```
# rpm -qR asbru-cm           
/bin/sh
/bin/sh
/bin/sh
/usr/bin/perl
bash
ftp
libwnck3
perl
perl(Carp)
perl(Compress::Raw::Zlib)
perl(Crypt::Blowfish)
perl(Data::Dumper)
perl(Digest::SHA)
perl(DynaLoader)
perl(Encode)
perl(Expect)
perl(Exporter)
perl(File::Basename)
perl(File::Copy)
perl(FindBin)
perl(Gtk3)
perl(Gtk3::SimpleList)
perl(HTTP::Proxy)
perl(IO::Handle)
perl(IO::Socket)
perl(IO::Socket::INET)
perl(MIME::Base64)
perl(Net::ARP)
perl(Net::Ping)
perl(OSSP::uuid)
perl(POSIX)
perl(Socket)
perl(Socket6)
perl(Storable)
perl(Sys::Hostname)
perl(Time::HiRes)
perl(XML::Parser)
perl(YAML)
perl(constant)
perl(lib)
perl(strict)
perl(utf8)
perl(vars)
perl(warnings)
perl-Crypt-CBC
perl-Crypt-Rijndael
perl-IO-Stty
perl-IO-Tty
perl-X11-GUITest
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsZstd) <= 5.4.18-1
telnet
vte
```

pygpgme seems to be unmaintained anyway, so hopefully the packagecloud installer is migrating away from it
https://src.fedoraproject.org/rpms/python-pygpgme

Even after removing the pygpgme package I have the same situation, so either I need to update -something- that the dependency tree is re-calculated(?) or there is another indirect dependency on python2

**Comment by egmontkob** [comment]
The `vte` requirement doesn't look okay, it's the GTK2 version, and indeed requires python2 as well. It should be `vte291`.

**Comment by gfrenoy** [comment]
Indeed!  Just discovered ```python2``` is required by ```vte``` !

Let me fix the rpm spec...

**Comment by gfrenoy** [comment]
Ok [it is fixed](https://github.com/asbru-cm/asbru-cm/commit/931c776b8b9c598b6b9a7d492dbe4a7053844a60).  I gave it a try on a fresh F31 live image, Ásbrú does not require python2 anymore :)

It will be available in 6.0.3 that will likely be released by the end of next week.  In the meantime, you can try the snapshot version, see [instructions](https://github.com/asbru-cm/asbru-cm/wiki/Installation-on-Fedora#snapshot-release).

Thanks for the heads-up !

**Comment by Croydon** [comment]
Thanks for patching it so quickly! 🎉
